pkg/util/push/otel.go

```diff
@@ -215,17 +258,80 @@ func TimeseriesToOTLPRequest(timeseries []prompb.TimeSeries) pmetricotlp.ExportR
 			attributes.PutStr(l.Name, l.Value)
 		}
 
-		metric := d.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
-		metric.SetName(name)
-		metric.SetEmptyGauge()
+		rm := d.ResourceMetrics()
+		sm := rm.AppendEmpty().ScopeMetrics()
+
+		if len(ts.Samples) > 0 {
+			metric := sm.AppendEmpty().Metrics().AppendEmpty()
+			metric.SetName(name)
+			metric.SetEmptyGauge()
+			for _, sample := range ts.Samples {
+				datapoint := metric.Gauge().DataPoints().AppendEmpty()
+				datapoint.SetTimestamp(pcommon.NewTimestampFromTime(time.Unix(0, int64(sample.Timestamp)*1000000)))
```

Comment:
nit: we can use `time.UnixMilli(sample.Timestamp)` or without converting to `time.Time`:  `pcommon.Timestamp(sample.Timestamp * time.Millisecond.Nanoseconds())`